deploy/crds/broker_v1alpha1_activemqartemis_crd.yaml

```diff
@@ -29,8 +29,12 @@ spec:
           properties:
             image:
               type: string
+            persistent:
+              type: boolean
             sslEnabled:
               type: boolean
+            clusterEnabled:
+              type: boolean
```

Comment:
Both in the case of clusterEnabled and others, there appears to be extra configuration that is required for the flag to take effect. It might be good to discuss and figure out if you really need these flags, and if not, provide a clear error and perhaps even fail deployment, if the extra property is not there. In other words, consider one of these options to follow best practices:

1- Remove clusterEnabled and look for the presence of clusterConfig to configure cluster
2- If clusterEnabled and clusterConfig are inconsistent (one set and the other not), fail deployment with clear error messages.

Reply:
Added related fields in CRD validation which would stop the CR from ever being created in the first place and adding extra validation with go code to check if sslEnabled/clusterEnabled and other parameters are missing failing the deployment with message for example in logs -"msg":" SSL Deployment Failed ","error":"SecretName is Required Field"